### PR TITLE
feat: add mail sending

### PR DESCRIPTION
**Pull request by Ismoil-JS**
only service part is needed for our project; the constroller is just written to check if the service is working or not
IMG_4326
IMG_4327
IMG_4328

### CODE REVIEW

File: src/modules/mail/mail.service.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      auth: {
+        user: this.configService.get('BREVO_LOGIN'),
+        pass: this.configService.get('BREVO_SMTP_KEY'),
+      },
+    });
+    this.bookingService = bookingService;
+  }
+
+  private formatBookingDate(date: Date): string {
+    return date.toLocaleDateString('uz-UZ', {
```
Comment (farrooh) [inline src/modules/mail/mail.service.ts:26]: Built-in Date is not really good, actually. You can use dayjs instead.

File: src/modules/mail/mail.service.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    `;
+
+    await this.sendEmail(subject, text, user.email);
+  }
+
+  async sendInvoiceMail(id: string) {
+    const booking = await this.bookingService.getBookingById(id);
+    const { user, accommodation, totalPrice } = booking;
+
+    const issueDate = new Date().toLocaleDateString('en-US');
```
Comment (farrooh) [inline src/modules/mail/mail.service.ts:88]: Use dayjs

File: docker-compose.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 services:
   postgres:
-    image: postgres
+    image: postgis/postgis
```
Comment (farrooh) [inline docker-compose.yml:3]: I don't think this change is related to the task.

File: src/modules/mail/mail.controller.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import { MailerService } from './mail.service';
+
+@Controller('mailer')
+export class MailerController {
+  constructor(private readonly mailerService: MailerService) {}
+
+  @Post('/:id')
+  async sendMail(@Param('id') id: string) {
+    return await this.mailerService.sendReviewMail(id);
+  }
```
Comment (farrooh) [inline src/modules/mail/mail.controller.ts:11]: Does this endpoint necessary? Shouldn't it be sent automatically after checkout?

File: src/modules/mail/mail.service.ts
Comment (farrooh) [inline src/modules/mail/mail.service.ts:1]: Add error handling to each method. Specify the type of the errors and where the error is being generated to make debugging easier. Use internal server error as the default error.

File: docker-compose.yml (hunk 3 above)
Comment (Ismoil-JS) [reply docker-compose.yml:3]: it is not related, but i was needed to chage it to run my code, it was not changed when location is added

File: src/modules/mail/mail.controller.ts (hunk 4 above)
Comment (Ismoil-JS) [reply src/modules/mail/mail.controller.ts:11]: it is not needed, i used it to check services

File: src/modules/mail/mail.module.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import { ConfigService } from '@nestjs/config';
+import { BookingsService } from '../bookings/bookings.service';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Booking } from '../bookings/entities/booking.entity';
+import { AccommodationsModule } from '../accommodations/accommodations.module';
+
+@Module({
+  controllers: [MailerController],
+  providers: [MailerService, ConfigService, BookingsService],
+  imports: [TypeOrmModule.forFeature([Booking]), AccommodationsModule],
```
[the remark this answers is not in the document]
Comment (Ismoil-JS) [reply src/modules/mail/mail.module.ts:13]: It includes accoommodation infos, so It is required, If I remove it or change it to Booking module, It is not working

**Review by farrooh — APPROVED** [review]

File: src/shared/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export * from './exceptions';
```
Comment (purple-mountain) [inline src/shared/index.ts:1]: can you delete this or put every other export here

**Review by purple-mountain — APPROVED** [review]

File: src/shared/index.ts (hunk 6 above)
Comment (Ismoil-JS) [reply src/shared/index.ts:1]: I can delete if others don't use it

**Pull request merged**